sdk/azidentity/confidential_client.go

```diff
@@ -23,8 +24,6 @@ import (
 )
 
 type confidentialClientOptions struct {
-	azcore.ClientOptions
```

Comment:
What's the rationale for unembedding this? Is it a matter of style?

Reply:
It seems like the `azcore.ClientOptions` are conceptually separate from the confidential and public client options which are MSAL specific, so I split them out. Purely cosmetic though, so I can revert this part if you prefer.

Reply:
I do prefer bundling all the options together because MSAL is an implementation detail of these wrapper types and the options are, at least in the abstract, all for the wrapper.

Reply:
Good point, fixed.